teuthology/orchestra/remote.py

```diff
@@ -206,6 +206,23 @@ def run(self, **kwargs):
         r.remote = self
         return r
 
+    def mkdtemp(self, suffix=None, parentdir=None):
+        """
+        Create a temporary directory on remote machine and return it's path.
+        """
+        args = ['mktemp', '-d']
+
+        if suffix is not None:
+            args.append('--suffix')
+            args.append(suffix)
+
+        if parentdir is not None:
+            args.append('-p')
+            args.append(parentdir)
+
+        proc = self.run(args=args, stdout=StringIO())
+        return proc.stdout.getvalue()
```

Comment:
You need to strip the output as well.

Reply:
`sh()` doesn't `strip()`; should I get go ahead and add that?

Reply:
yep

Reply:
Oops, I thought you meant adding strip in mkdtemp. Sorry for this, we do not need to strip inside `sh()`. Apologize.